Src/GoogleApis.Auth/OAuth2/AuthorizationCodeInstalledApp.cs

```diff
@@ -88,7 +87,18 @@ public ICodeReceiver CodeReceiver
                     taskCancellationToken).ConfigureAwait(false);
             }
 
-            return new UserCredential(flow, userId, token);
+            return new UserCredential(Flow, userId, token);
+        }
+
+        /// <summary>
+        /// Determines the need for retrieval of a new authorization code, based on the given token and the 
+        /// authorization code flow.
+        /// </summary>
+        public bool ShouldRequestAuthorizationCode(TokenResponse token)
+        {
+            // If the flow includes a parameter that requires a new token, if the stored token is null or it doesn't
+            // have a refresh token and the access token is expired we need to retrieve a new authorization code.
+            return Flow.ShouldForceTokenRetrieval() || token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock));

```

Comment:
It was, thank you. Fixed. I broke the line before the operand of && so that a glance at the next line shows it's part of the previous line's operation, I'm not sure if there is some best practice I should otherwise be adhering to there. If you'd rather the line break be after the operand before the variable let me know.